### PR TITLE
Modernize source code: last part

### PR DESCRIPTION
**Pull request by nerzhul**
* Use empty when needed
* Use emplace_back instead of push_back when needed
* For range-based loops
* Initializers fixes
* constructors, destructors default
* c++ C stl includes

### CODE REVIEW

File: src/voxelalgorithms.cpp [hunk 1]
```diff
@@ -265,8 +260,7 @@ struct LightQueue {
 		direction source_dir)
 	{
 		assert(light <= LIGHT_SUN);
-		lights[light].push_back(
-			ChangingLight(rel_pos, block_pos, block, source_dir));
+		lights[light].emplace_back(rel_pos, block_pos, block, source_dir);
```
Comment (paramat) [inline src/voxelalgorithms.cpp]: `lights[light].emplace_back(ChangingLight(rel_pos, block_pos, block, source_dir));`
?

Comment (paramat) [inline src/voxelalgorithms.cpp]: Hmm actually this seems intentional, nevermind.

**Comment by paramat** [comment]
Mapgen stuff seems ok.

**Comment by nerzhul** [comment]
When you push back an object constructor you must use emplace_back to push + create to prevent an object copy

File: src/database-dummy.h [hunk 2]
```diff
@@ -35,7 +35,7 @@ class Database_Dummy : public MapDatabase, public PlayerDatabase
 	void savePlayer(RemotePlayer *player) {}
 	bool loadPlayer(RemotePlayer *player, PlayerSAO *sao) { return true; }
 	bool removePlayer(const std::string &name) { return true; }
-	void listPlayers(std::vector<std::string> &) {}
+	void listPlayers(std::vector<std::string> & res) {}
```
Comment (SmallJoker) [inline src/database-dummy.h]: Superfluous space.

File: src/script/cpp_api/s_inventory.cpp [hunk 3]
```diff
@@ -230,20 +230,17 @@ bool ScriptApiDetached::getDetachedInventoryCallback(
...
-		errorstream<<"Detached inventory \""<<name<<"\" callback \""
-			<<callbackname<<"\" is not a function"<<std::endl;
+
+	if (lua_isnil(L, -1)) {
 		lua_pop(L, 1);
 		return false;
 	}
+
+	errorstream<<"Detached inventory \""<<name<<"\" callback \""
+		<<callbackname<<"\" is not a function"<<std::endl;
```
Comment (SmallJoker) [inline src/script/cpp_api/s_inventory.cpp]: Spaces missing.

File: src/script/lua_api/l_inventory.cpp [hunk 4]
```diff
@@ -500,23 +500,24 @@ int ModApiInventory::l_get_inventory(lua_State *L)
...
+	NO_MAP_LOCK_REQUIRED;
+	if (type == "player") {
+		std::string name = checkstringfield(L, 1, "name");
+		loc.setPlayer(name);
+	} else if(type == "detached") {
+		std::string name = checkstringfield(L, 1, "name");
+		loc.setDetached(name);
 	}
+
+	if(getServer(L)->getInventory(loc) != NULL)
```
Comment (SmallJoker) [inline src/script/lua_api/l_inventory.cpp]: Space after `if`, same for the line `"detached"` above.

File: src/tool.cpp [hunk 5]
```diff
@@ -159,14 +156,13 @@ HitParams getHitParams(const ItemGroupList &armor_groups,
 	s16 damage = 0;
 	float full_punch_interval = tp->full_punch_interval;
 
-	for (DamageGroup::const_iterator i = tp->damageGroups.begin();
-			i != tp->damageGroups.end(); ++i) {
-		s16 armor = itemgroup_get(armor_groups, i->first);
-		damage += i->second * rangelim(time_from_last_punch / full_punch_interval, 0.0, 1.0)
+	for (const auto &damageGroup : tp->damageGroups) {
+		s16 armor = itemgroup_get(armor_groups, damageGroup.first);
+		damage += damageGroup.second * rangelim(time_from_last_punch / full_punch_interval, 0.0, 1.0)
```
Comment (SmallJoker) [inline src/tool.cpp]: Line too long.

**Pull request merged**